WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -181,6 +181,11 @@ + (NSArray *)routes
   NSInteger elementID = [request.parameters[@"id"] integerValue];
   XCUIElement *element = [elementCache elementForIndex:elementID];
   NSError *error = nil;
+  if (!element.fb_isVisible) {
+    if (![element fb_scrollToVisibleWithError:&error]) {
```

Comment:
Yep, but this implementation will not work, because during scroll element "under" XCUIElement may change. So you might end up tapping wrong element. I suspect that this has something to do with cell reusability. As workaround we don't "autoscroll" when tapping yet and shift scrolling responsibility and identifying correct item to the client. I plan to work on this issue in a wink. I opened [issue](https://github.com/facebook/WebDriverAgent/issues/157) to track it.